query-engine/prisma-models/src/model.rs

```diff
@@ -74,6 +85,10 @@ impl Model {
         self.manifestation.as_ref().map(|m| m.as_ref())
     }
 
+    pub fn find_relation(&self, relation_name: &str) -> crate::Result<RelationWeakRef> {
+        self.internal_data_model().find_relation(relation_name)
+    }
```

Comment:
This mixes abstractions and can lead to nasty bugs. It looks for a relationship in the entire data model, but here we're in the `Model` abstraction. Point being, I'd expect this function to fail if you look for a relationship that doesn't belong to this model. But this _won't_.

Either you meant to look in `self.fields()` or it should just not exist at all.

Looking at how this function is used in this PR, eg: 👇

```diff
    pub fn relation(&self) -> RelationRef {
        self.relation
            .get_or_init(|| {
-                self.model()
-                   .internal_data_model()
-                  .find_relation(&self.relation_name)
-                    .unwrap()
-            })
+            .get_or_init(|| self.model().find_relation(&self.relation_name).unwrap())
```

This method should not exist at all in `Model`. It's important we don't blindly blur the abstractions for the sake of reducing code boilerplate. It makes the code slightly more painful to write but also often easier to read because you can easily follow where things are coming from.

Reply:
This is a good spot. I didn't think of it that way.